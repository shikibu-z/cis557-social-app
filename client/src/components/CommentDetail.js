/* eslint-disable eqeqeq */
import React, { useState, useEffect } from 'react';
import '../stylesheets/CommentDetail.css';

const lib = require('../api/fetchers');

function CommentDetail(props) {
  const userid = localStorage.getItem('currUser');
  const [commentAuthor, setCommentAuthor] = useState();
  const [isLoading, setIsLoading] = useState(true);
  const {
    author, timestamp, content, groupid, postid, commentid,
  } = props;

  function handleDeleteComment() {
    lib.deleteComment(groupid, postid, commentid);
    window.location.reload();
  }

  useEffect(async () => {
    const authorRes = await lib.fetchUserProfile(author);
    setCommentAuthor(authorRes.data);
    setIsLoading(false);
  }, []);
  return (
    !isLoading
    && (
    <div className="commentWrapper">
      <img src={commentAuthor.photo} alt="comment user" />
      <div className="commentDetailWrapper">
        <div className="userInfoWrapper">
          <div id="commentUserName">{commentAuthor.username}</div>
          <div id="timeFromNow">{timestamp}</div>
          {userid == author && <button type="submit" id="deleteOption" onClick={handleDeleteComment}>delete</button>}
        </div>
        <div id="commentBody">{content}</div>
      </div>
    </div>
    )
  );
}

export default CommentDetail;
