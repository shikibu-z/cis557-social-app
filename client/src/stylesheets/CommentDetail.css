.commentWrapper{
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  margin: 1% 2% 1% 2%;
}

.commentWrapper img{
  display: block;
  width: 40px;
  height: 40px;
  border-radius: 50%;
  overflow: hidden;
  margin: 1%;
}

.commentDetailWrapper{
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

.userInfoWrapper{
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: baseline;
  width: fit-content;
}

#commentBody {
  width: fit-content;
  height: fit-content;
}

#commentUserName{
  white-space:nowrap;
  width: fit-content;
  font-family: Roboto;
  font-size: 22px;
  font-style: normal;
  font-weight: bold;
  letter-spacing: 0px;
  text-align: center;
  color: #f8b114;
  margin-right: 4%;
}

#timeFromNow{
  white-space:nowrap;
  width: fit-content;
  font-family: Roboto;
  font-size: 14px;
  font-style: normal;
  letter-spacing: 0px;
  text-align: center;
  margin-right: 4%;
}

#editOption{
  white-space: nowrap;
  width: fit-content;
  font-family: Roboto;
  font-size: 12px;
  font-style: normal;
  letter-spacing: 0px;
  text-align: center;
  color: cornflowerblue;
  margin-right: 2%;
  background-color: white;
  border: none;
}

#deleteOption{
  white-space: nowrap;
  font-family: Roboto;
  font-size: 12px;
  font-style: normal;
  letter-spacing: 0px;
  text-align: center;
  color: cornflowerblue;
  margin-right: 2%;
  background-color: white;
  border: none;
}