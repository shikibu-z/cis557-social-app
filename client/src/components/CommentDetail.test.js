/*eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import CommentDetail from './CommentDetail';

test('Comment Detail matches snapshot', () => {
  const component = renderer.create(<CommentDetail />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
