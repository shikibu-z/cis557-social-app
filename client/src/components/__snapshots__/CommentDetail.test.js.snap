// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Detail matches snapshot 1`] = `null`;
